components/common/infinite_scroll.jsx

```diff
@@ -0,0 +1,155 @@
+
+        /**
+         * The number of items to load in a single fetch
+         */
+        itemsPerPage: PropTypes.number.isRequired,
+
+        /**
+         * The current page that has been scrolled to
+         */
+        pageNumber: PropTypes.number.isRequired,
+
+        /**
+         * Optional style object that's passed on to the underlying loader
+         * component
+         */
+
+        loaderStyle: PropTypes.object,
+    };
+
+    static defaultProps = {
+        bufferValue: SCROLL_BUFFER,
+        endOfDataMessage: '',
+        styleClass: '',
+        loaderStyle: {},
+    };
+
+    constructor(props) {
+        super(props);
+        this.state = {
+            isFetching: false,
+            isEndofData: false,
+        };
+        this.node = React.createRef();
+    }
+
+    componentDidMount() {
+        this.node.current.addEventListener('scroll', this.debounceHandleScroll);
+    }
+
+    componentWillUnmount() {
+        this.node.current.removeEventListener('scroll', this.debounceHandleScroll);
+    }
+
+    validateBuffer = (buffer) => {
+        if (buffer < SCROLL_BUFFER) {
+            return SCROLL_BUFFER;
+        }
+        return Math.abs(buffer);
+    }
+
+    getAmountOfPages = (total, freq) => {
+        return Math.ceil(total / freq);
+    }
+
+    handleScroll = () => {
+        const {isFetching, isEndofData} = this.state;
+        const {callBack, bufferValue, totalItems, itemsPerPage, pageNumber} = this.props;
+
+        const node = this.node.current;
+        const validBuffer = this.validateBuffer(bufferValue);
+
+        const toScroll = node.scrollHeight - node.clientHeight - validBuffer;
+        const nearBottom = node.scrollTop > toScroll;
+
+        if (nearBottom && !isEndofData && !isFetching) {
+            this.setState({isFetching: true},
+                async () => {
+                    await callBack();
+
+                    this.setState({
+                        isFetching: false,
+                    });
+
+                    if (totalItems === 0) {
+                        return;
+                    }
+
+                    const amountOfPages = this.getAmountOfPages(totalItems, itemsPerPage);
+
+                    if (pageNumber === amountOfPages) {
+                        this.setState({
+                            isEndofData: true,
+                        });
+                    }
+                });
+        }
+    }
+
+    debounceHandleScroll = debounce(this.handleScroll, DEBOUNCE_WAIT_TIME);
+
+    render() {
+        const {children, endOfDataMessage, styleClass, loaderStyle} = this.props;
+        const {isFetching, isEndofData} = this.state;
+        return (
+            // eslint-disable-next-line react/jsx-filename-extension
+            <>
+                <div
+                    className={`infinite-scroll ${styleClass}`}
+                    ref={this.node}
+                >
+                    {children}
+                    {(isFetching) && (
```

Comment:
To do the change of showing loader by default. i think all we have to do is change this condition 

`isFetching` to `!isEndofData` so we dont show it if there is no more data to show. Though we need assign the state on first load as well so we don't have a loader at the bottom if all the available teams(less than 30) are returned on first API call of get teams.

Reply:
@sudheerDev  thanks for the review. Working on this right now. I got a little confused at first with what you meant "having loader as part of of the list". I think this makes it more clear. Thanks.